File: cdn-system/api/models/domain_origin.go
package models

import "time"

type DomainOrigin struct {
	ID        int64     `json:"id" gorm:"primaryKey"`
	DomainID  int64     `json:"domain_id" gorm:"index"`
	Addr      string    `json:"addr"`
	Port      int       `json:"port"`
	Weight    int       `json:"weight"`
	Protocol  string    `json:"protocol"` // http or https
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: cdn-system/cnn.net/src/Cnn.Common/Domain/Entities/DomainOrigin.cs
using SqlSugar;

namespace Cnn.Domain.Entities;

[SugarTable("domain_origins")]
public class DomainOrigin
{
    [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
    public long Id { get; set; }

    public long? DomainId { get; set; }

    public string? Addr { get; set; }

    public int? Port { get; set; }

    public int? Weight { get; set; }

    public string? Protocol { get; set; }

    [SugarColumn(ColumnName = "created_at")]
    public DateTime? CreatedAt { get; set; }

    [SugarColumn(ColumnName = "updated_at")]
    public DateTime? UpdatedAt { get; set; }
}
